src/wp-admin/includes/class-wp-automatic-updater.php

```diff
@@ -1540,6 +1540,16 @@ protected function send_plugin_theme_email( $type, $successful_updates, $failed_
 		 */
 		$email = apply_filters( 'auto_plugin_theme_update_email', $email, $type, $successful_updates, $failed_updates );
 
+		if ( defined( 'WP_DEBUG' ) && WP_DEBUG && 'fail' === $type ) {
```

Comment:
The constant is always defined in `default-constants.php` so the initial check can be removed:

```suggestion
		if ( WP_DEBUG && 'fail' === $type ) {
```